R/convertREN21.R

```diff
@@ -5,242 +5,230 @@
+    x_cap_tic[, , "Hydro"] <- pmax(x_cap[, , "Hydro"], x_tmp[, target_years, "TIC-Absolute.Hydro", drop = TRUE] * setYears(cf_hydro[getItems(x_tmp,dim=1), , ] * 8760))
 
     # Converting Production targets (GWh) to Capacity targets (TIC-Absolute) (GW) for geothermal and biomass
     # pmax takes the higher value from existing capacity and new capacity derived (from production)
-    x_cap_pt[,,"Biomass"] <- pmax(x_cap[,,"Biomass"],x_tmp[,target_years,"Production-Absolute.Biomass"]/(8760*cf_biomass))
-    
-    x_cap_pt[,,"Geothermal"] <- pmax(x_cap[,,"Geothermal"],x_tmp[,target_years,c("Production-Absolute.Geothermal")]/(8760*cf_geothermal))    
+    x_cap_pt[, , "Biomass"] <- pmax(x_cap[, , "Biomass"], x_tmp[, target_years, "Production-Absolute.Biomass"] / (8760 * cf_biomass))
+
+    x_cap_pt[, , "Geothermal"] <- pmax(x_cap[, , "Geothermal"], x_tmp[, target_years, c("Production-Absolute.Geothermal")] / (8760 * cf_geothermal))
   # for solar, wind, and hydro conversion will be done using another method below
-  x_cap_pt[,target_years,c("SolarPV","SolarCSP","Wind")] <- x_tmp[,target_years,c("SolarPV","SolarCSP","Wind")][,,"Production-Absolute",drop=TRUE]
-  
-  
+  x_cap_pt[, target_years, c("SolarPV", "SolarCSP", "Wind")] <- x_tmp[, target_years, c("SolarPV", "SolarCSP", "Wind")][, , "Production-Absolute", drop = TRUE]
+
+
 # Converting Production targets to capacity targets for solar and wind
 # Fpr Hydro, all targets (which have already been converted into generation targets) will be converted into capacity targets based on maxprod and capacity factors.
 #
 
   # x_tmp[,target_years,"Production-Absolute.Hydro"] <- pmax(x_tmp[,target_years,"Production-Absolute.Hydro"],x_cap_tic[,,"Hydro"],x_cap_ac[,,"Hydro"])
   # Special case for hydro as mentioned above
-  x_cap_pt[,target_years,"Hydro"] <- pmax(x_tmp[,target_years,"Production-Absolute.Hydro"],x_cap_tic[,,"Hydro"],x_cap_ac[,,"Hydro"])
+  x_cap_pt[, target_years, "Hydro"] <- pmax(x_tmp[, target_years, "Production-Absolute.Hydro"], x_cap_tic[, , "Hydro"], x_cap_ac[, , "Hydro"])
 
 # Obtaining the capacity factors (nur) values and associated maxproduction (maxprod) for Hydro, Wind, and Solar
 
     data_wind <- calcOutput("PotentialWindOn", aggregate = FALSE)
     # Reordering dim=3 for data_wind so that 1st position corresponds to maxprod.nur.1 and not maxprod.nur.9
-    data_wind_sorted <- mbind(data_wind[,,"1"],data_wind[,,"2"],data_wind[,,"3"],data_wind[,,"4"],
-                              data_wind[,,"5"],data_wind[,,"6"],data_wind[,,"7"],data_wind[,,"8"],data_wind[,,"9"])
+    data_wind_sorted <- mbind(data_wind[, , "1"], data_wind[, , "2"], data_wind[, , "3"], data_wind[, , "4"],
+                              data_wind[, , "5"], data_wind[, , "6"], data_wind[, , "7"], data_wind[, , "8"], data_wind[, , "9"])
     data_hydro <- calcOutput("PotentialHydro", aggregate = FALSE)
     #data_solar <- calcOutput("Solar", aggregate = FALSE)
     data_solar <- calcOutput("Solar")
```

Comment:
you are still using solar data on the regional instead of country level. Is this correct?